### PR TITLE
WIP: using formatter.nvim instead of neoformat

### LINKED ISSUE #775: Change formatter to formatter.nvim
https://github.com/mhartington/formatter.nvim

### PR DESCRIPTION
**Pull request by abzcoding**
Fixes #775

we can have some defaults for various langs, for example in `lua/default-config.lua`
```lua
rust = {
-- ...
      formatter = {
        exe = "rustfmt",
        args = {"--emit=stdout"},
      },
-- ...
},
javascript = {
-- ...
      formatter = {
        exe = "prettier",
        args = {"--stdin-filepath", vim.api.nvim_buf_get_name(0), '--single-quote'},
-- ...
}
```

users can change the default formatter by having this in their `lv-config.lua`:
```lua
-- rust
O.lang.rust.formatter = {
  exe = "rustfmt",
  args = {"--emit=stdout"},
}
```

and you can disable formatting for a lang by setting their `formatter` to an empty array inside `lv-config.lua`:
```lua
-- disable php formatting
O.lang.php.formatter = {}
```

I kept this `WIP` to get suggestions about what default formatter we want.

### CODE REVIEW

**Comment by rafamadriz** [comment]
May I suggest that if you're gonna go through every single formatter and create the configuration for [formatter.nvim](https://github.com/mhartington/formatter.nvim), first workout this as a PR to the formatter.nvim guys ?? The would make so everyone *(not just people using LunarVim)* can benefit from having everything set up with formatter.nvim.

**Comment by ChristianChiarulli** [comment]
@rafamadriz that's good idea.

**Comment by ChristianChiarulli** [comment]
For defaults as long as something is there and working I'm not too worried about it. I'm sure issues will role in complaining and we can update then. Only one I care about ie `stylua` should be the default for lua.

**Comment by abzcoding** [comment]
> For defaults as long as something is there and working I'm not too worried about it. I'm sure issues will role in complaining and we can update then. Only one I care about ie `stylua` should be the default for lua.

@ChristianChiarulli  can you test if the `lua` formatting works to your liking or not? possibly on tonight's stream :wink:

File: lua/lv-formatter/init.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    end
+  end
+end
+require("formatter").setup {
+  logging = false,
+  filetype = formatter_filetypes,
+}
+
+if not O.format_on_save then
+  vim.cmd [[if exists('#autoformat#BufWritePre')
```
[the remark this answers is not in the document]
Comment (abzcoding) [reply lua/lv-formatter/init.lua]: nice catch 👀 , fixed it

**Pull request merged**